### PR TITLE
[OptTable] Make new lines in help text respect indentation

### PR DESCRIPTION
**Pull request by avillega**
With this changes, new lines in the HelpText defined
in OptTable have the same indentation as the first line.

Before, the help output will look something like:

```
--color=<value>       Whether to use color when
symbolizing log markup: always, auto, never
```

With this change:

```
--color=<value>       Whether to use color when
                      symbolizing log markup: always, auto, never
```

Note the "-" at the beginning of the help text in the new version,
it was added to clearly separate the HelpText of the different
options.

### CODE REVIEW

**Comment by MaskRay** [comment]
Thanks. This is an improvement.

Some tests may need `FileCheck --match-full-lines --strict-whitespace` to test the whitespace change
https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-FileCheck-strict-whitespace

Sometimes we switch to the following check prefix style:
```
#      CHECK:xxx
# CHECK-NEXT:xxx
# CHECK-NEXT:xxx
```

**Review by jh7370 — COMMENTED** [review]
Is the main functionality of this test actually tested in the current patch? It doesn't seem to be to me.

**Comment by avillega** [comment]
> Is the main functionality of this test actually tested in the current patch? It doesn't seem to be to me.

I don't fully understand your comment, I am working on getting some of the test to be whitespace strict to specifically test the functionality.

File: llvm/test/tools/llvm-cvtres/help.test [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-; HELP_TEST: 	  OVERVIEW: Resource Converter
-; HELP_TEST-DAG:  USAGE: llvm-cvtres [options] file...
-; HELP_TEST-DAG:  OPTIONS:
-; HELP_TEST-NEXT:   /{{DEFINE}}:symbol - Not implemented
-; HELP_TEST-NEXT:   /FOLDDUPS: - Not implemented
-; HELP_TEST-NEXT:   /HELP - Display available options
+;      HELP_TEST: OVERVIEW: Resource Converter
+;  HELP_TEST-DAG: USAGE: llvm-cvtres [options] file...
+;  HELP_TEST-DAG: OPTIONS:
+; HELP_TEST-NEXT:   /{{DEFINE}}:symbol     - Not implemented
```
Comment (jh7370) [inline llvm/test/tools/llvm-cvtres/help.test:7]: The formatting in this line looks wrong? It looks to me like there might be a bug here (possibly predating your PR).

**Review by jh7370 — COMMENTED** [review]
> > Is the main functionality of this test actually tested in the current patch? It doesn't seem to be to me.
> I don't fully understand your comment, I am working on getting some of the test to be whitespace strict to specifically test the functionality.

I just didn't see any testing like you've now added in OptionParsingTest.cpp that showed that the indentation was working correctly as desired.

Taking a step back, I'm not sure the addition of the dash has anything to do with the main purpose of this PR, and I'm a little worried people will not spot that this PR is adding it. I personally have no issue with that part of the change, but I wonder if others wouldn't want it?

File: llvm/test/tools/llvm-cvtres/help.test (hunk 1 above)
Comment (avillega) [reply llvm/test/tools/llvm-cvtres/help.test:7]: This is a quirk of FileCheck. It thinks that `DEFINE` is a directive, so in order to match the `DEFINE` word it is necessary to make it a regex. FileCheck will ignore `{{}}` and that is why it doesn't look at the same indentation level as the rest of the options, but it is.

**Comment by avillega** [comment]
> > Note the "-" at the beginning of the help text in the new version,
> > it was added to clearly separate the HelpText of the different
> > options.
> 
> > Taking a step back, I'm not sure the addition of the dash has anything to do with the main purpose of this PR, and I'm a little worried people will not spot that this PR is adding it. I personally have no issue with that part of the change, but I wonder if others wouldn't want it?
> 
> I think `-` in `--help` messages is (very) uncommon. Many programs (`cat, node, gcc, as, 7z, ...`) don't use `-`.

I will remove it. But I added it for two reasons:
- I wanted a way to clearly spot the beginning of the help text of an option given that new lines will be indented at the same level as the first line of the help text, it is more difficult to clearly see the start and end of the help text for an specific option.
- The help text generated by llvm::cl uses the `- ` to separate the help text of each option so I was trying to do something similar to what many llvm tools still do.

**Review by jh7370 — COMMENTED** [review]
I don't know if this is due to your use of spr or something, but the rebasing has made this change pretty much impossible to review, because it has introduced a whole load of changes that have nothing to do with the original patch, and there's no longer a convenient commit at the end of the stack that I can review.

**Comment by avillega** [comment]
> I don't know if this is due to your use of spr or something, but the rebasing has made this change pretty much impossible to review, because it has introduced a whole load of changes that have nothing to do with the original patch, and there's no longer a convenient commit at the end of the stack that I can review.

The github workflow is a pain, and spr is not making it any better, at least for this specific PR. I created another PR with the same changes and added you as reviewers. New PR at: https://github.com/llvm/llvm-project/pull/75366

**Comment by MaskRay** [comment]
> > I don't know if this is due to your use of spr or something, but the rebasing has made this change pretty much impossible to review, because it has introduced a whole load of changes that have nothing to do with the original patch, and there's no longer a convenient commit at the end of the stack that I can review.
> 
> The github workflow is a pain, and spr is not making it any better, at least for this specific PR. I created another PR with the same changes and added you as reviewers. New PR at: #75366

It seems spr that a few contributors are using does solve my complaint about the merge-upstream workflow (https://maskray.me/blog/2023-09-09-reflections-on-llvm-switch-to-github-pull-requests#patch-evolution)

```
git switch pr1
git merge origin/main
git push $username pr1
```

It seems that reviewers need to learn to ignore these spr-generated rebase commits.
I want to try spr. Amend+rebase is just natural.

**Comment by jh7370** [comment]
> > > I don't know if this is due to your use of spr or something, but the rebasing has made this change pretty much impossible to review, because it has introduced a whole load of changes that have nothing to do with the original patch, and there's no longer a convenient commit at the end of the stack that I can review.
> > 
> > 
> > The github workflow is a pain, and spr is not making it any better, at least for this specific PR. I created another PR with the same changes and added you as reviewers. New PR at: #75366
> 
> It seems spr that a few contributors are using does solve my complaint about the merge-upstream workflow (https://maskray.me/blog/2023-09-09-reflections-on-llvm-switch-to-github-pull-requests#patch-evolution)
> 
> ```
> git switch pr1
> git merge origin/main
> git push $username pr1
> ```
> 
> It seems that reviewers need to learn to ignore these spr-generated rebase commits. I want to try spr. Amend+rebase is just natural.

I tried clicking on the option to review the specific commit that actually contained changes, but it also always selected the immediately-previous spr commit that included a rebase, and as such it became impossible to review individually. More generally, it's no longer possible to review just the actual changes as a whole (ignoring things introduced via rebase), as far as I can tell.

**Pull request closed without merging**